### DISCUSSION TITLE
Warp Charge/other proprietary USB power protocol support

### DISCUSSION THREAD

**Discussion by adryzz** [Ralim/IronOS#1707] [Ideas]
Since the USB PD/QC stack is all software, would it be possible to support something like Warp Charge, to allow using an iron with the same power brick as a phone, if this doesn't support PD/QC? If not, what would be the showstoppers?

**Comment by gamelaster** [comment]
I think it should be possible, the biggest stopper is that someone needs to implement it, and protocols to communicate with those chargers are not always open, or if they are, they are complex or not much documented.

**Reply by adryzz** [reply]
yeah, exactly, most of these should be proprietary and unknown. VOOC/Warp Charge appears to have been implemented by LineageOS here https://github.com/LineageOS/android_kernel_oppo_msm8974/tree/lineage-16.0/drivers/power/oppo

**Comment by Ralim** [comment]
It is possible _if_ we have the hardware for it. OPPO/WARP are not possible as we dont have the hardware to talk the protocol on the extra data pin it communicates on.
